pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -302,6 +301,21 @@ public static int getBatchSizeforEntry(ByteBuf metadataAndPayload, Subscription
         return -1;
     }
 
+    public static MessageMetadata peekMessageMetadata(ByteBuf metadataAndPayload, Subscription subscription,
```

Comment:
Just a small hint, maybe we could refactor ` getBatchSizeforEntry()` in order to use this method.

Reply:
Very good point.. fixing this